server/service/src/document/patient/insert.rs

```diff
@@ -0,0 +1,82 @@
+use chrono::Utc;
+use repository::{EqualFilter, RepositoryError};
+use util::uuid::uuid;
+
+use crate::{
+    document::{
+        document_service::{DocumentInsertError, DocumentService, DocumentServiceTrait},
+        raw_document::RawDocument,
+    },
+    service_provider::ServiceContext,
+};
+
+use super::{
+    patient_doc_name, patient_schema::SchemaPatient, Patient, PatientFilter, PatientService,
+    PatientServiceTrait, PATIENT_TYPE,
+};
+
+#[derive(PartialEq, Debug)]
+pub enum InsertPatientError {
+    InvalidDataSchema(Vec<String>),
+    InternalError(String),
+    DatabaseError(RepositoryError),
+}
+
+pub struct InsertPatient {
+    pub data: serde_json::Value,
+    pub schema_id: Option<String>,
+}
+
+pub fn insert_patients(
```

Comment:
would be good when you implement the document schemas and forms to have a seperate endpoint to get patient insert form (for ease of use by front end)